pyglmnet/pyglmnet.py

```diff
@@ -8,16 +8,47 @@
 from scipy.stats import norm
 
 from .utils import logger, set_log_level, _check_params, \
-    _verbose_iterable, _tqdm_log, softplus
+    _verbose_iterable, _tqdm_log
 from .base import BaseEstimator, is_classifier, check_version
 
 from .externals.sklearn.utils import check_random_state, check_array, check_X_y
 from .externals.sklearn.utils.validation import check_is_fitted
 
+from .distributions import BaseDistribution, Gaussian, Poisson, \
+    PoissonSoftplus, NegBinomialSoftplus, Binomial, Probit, GammaSoftplus, \
+    softplus
+
 ALLOWED_DISTRS = ['gaussian', 'binomial', 'softplus', 'poisson',
                   'probit', 'gamma', 'neg-binomial']
 
 
+def _get_distr(distr):
+    if isinstance(distr, BaseDistribution):
+        return distr
+
+    elif distr == 'gaussian':
```

Comment:
I wonder if we can change this huge if-else to something in which we dynamically instantiate the correct class given the string. I was thinking of something like this https://softwareengineering.stackexchange.com/a/351394, in which we have a dictionary which is holding all the required classes.